pkg/controller/clusterdeployment/clusterprovisions.go

```diff
@@ -220,6 +234,39 @@ func (r *ReconcileClusterDeployment) startNewProvision(
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileClusterDeployment) shouldRetryBasedOnFailureReason(prov *hivev1.ClusterProvision, logger log.FieldLogger) (bool, error) {
+	// prov will be nil if there have been no failed provisions yet
+	if prov == nil {
+		logger.Debug("no failed provisions yet -- allowing retry")
+		return true, nil
+	}
+	// Load up HiveConfig
+	hc := &hivev1.HiveConfig{}
+	if err := r.Get(context.TODO(), types.NamespacedName{Name: constants.HiveConfigName}, hc); err != nil {
+		return false, err // caller logs
```

Comment:
Is this a common pattern in this repo? I was under the impression that we let the hive-operator read HiveConfig and pass any desired configuration into the hive-controllers Pod, but that hive-controllers wouldn't be reading the HiveConfig directly.

Reply:
depending on the info being passed, it can be as simple as env vars, or a dedicated cli parameter for the hive-controller binary, or as complex as passing in a configmap. for example the managedns setup uses a ConfigMap https://github.com/openshift/hive/blob/master/pkg/operator/hive/manageddns.go

Reply:
I guess in this case a configmap is appropriate since it's complex data (a list of strings).